packages/chrome/src/ChromeProvider/ChromeProvider.tsx

```diff
@@ -1,78 +1,69 @@
-  const scalprum = useScalprum<{ initialized: boolean; api: { chrome: ChromeAPI } }>();
-  const { pathname } = useLocation();
-  const postData = async (pathname: string, title: string, bundle: string) => {
+  const inactiveDuration = 1000 * 20;
+  const batchUpdateInterval = 1000 * 60 * 3;
+  // There was some lint and namespacing weirdness with ChromeAPI and NodeJS.Timeout so
+  // we're grabbing the type without managing another import or adding anything to tsconfig.
+  const inactiveTimeout = useRef<ReturnType<typeof setTimeout> | undefined>(undefined);
+  const postBatchData = async (recentPages: LastVisitedPage[]) => {
     try {
-      const data = await postDataDebounced(pathname, title, bundle);
-      if (data) {
-        providerState.setLastVisited(data);
-      }
+      await post<LastVisitedPage[], { pages: LastVisitedPage[] }>(LAST_VISITED_URL, {
+        pages: recentPages,
+      });
     } catch (error) {
       console.error('Unable to update last visited pages!', error);
     }
   };
+
+  const sanitizeRecentPages = (recentPages: LastVisitedPage[]): LastVisitedPage[] => {
+    return recentPages.map(({ pathname, title, bundle }) => ({ pathname, title, bundle }));
+  };
+
+  const updateMostRecentPages = async () => {
+    const recentPages = providerState.getState().lastVisitedPages;
+    const sanitizedPages: LastVisitedPage[] = sanitizeRecentPages(recentPages);
+    await postBatchData(sanitizedPages);
+  };
+
   useEffect(() => {
-    let titleObserver: MutationObserver | undefined;
-    let prevTitle: string | null;
-    const titleTarget = document.querySelector('title');
-    if (titleTarget) {
-      prevTitle = titleTarget.textContent;
-      // initial api call on mount
-      postData(pathname, prevTitle ?? '', scalprum.api.chrome.getBundleData().bundleTitle);
-      /**
-       * Use Mutation observer to trigger the updates.
-       * Using the observer will ensure the last visited pages gets updated on document title change rather than just location change.
-       * The chrome service uses pathname as identifier and updates title according.
-       * Multiple calls with the same pathname and different title will ensure that the latest correct title is assigned to a pathname.       *
-       *  */
-      titleObserver = new MutationObserver((mutations) => {
-        // grab text from the title element
-        const currentTitle = mutations[0]?.target.textContent;
-        // trigger only if the titles are different
-        if (typeof currentTitle === 'string' && currentTitle !== prevTitle) {
-          try {
-            prevTitle = currentTitle;
-            postData(pathname, currentTitle, scalprum.api.chrome.getBundleData().bundleTitle);
-          } catch (error) {
-            // catch sync errors
-            console.error('Unable to update last visited pages!', error);
-          }
+    // Save state from localStorage on an interval
+    setInterval(async () => {
```

Comment:
You have to clear the interval in the useEffect cleanup phase. Currently this can cause memory leaks.